### PR TITLE
CB-15245:Fix NPE in create-aws-cluster when encryption type set to null

### PR DESCRIPTION
**Pull request by satyabollineni**
Tested this with a custom template where encryption type is set to null and DH creation success.

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/InstanceTemplateParameterConverter.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ private AwsInstanceTemplateV4SpotParameters convert(AwsInstanceTemplateV1SpotPar
     private AwsEncryptionV4Parameters convert(AwsEncryptionV1Parameters source, DetailedEnvironmentResponse environment) {
         AwsEncryptionV4Parameters response = new AwsEncryptionV4Parameters();
         EncryptionType dataHubEncryptionKeyType = source.getType();
-        if (dataHubEncryptionKeyType.equals(EncryptionType.CUSTOM)) {
+        if (dataHubEncryptionKeyType != null && dataHubEncryptionKeyType.equals(EncryptionType.CUSTOM)) {
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/InstanceTemplateParameterConverter.java:59]: No, the usual way to do this is swapping the objects around `equals()`. So
```
EncryptionType.CUSTOM.equals(dataHubEncryptionKeyType)
```
That construct will never result in an NPE.

**Review by lajosrodek — CHANGES_REQUESTED** [review]

**Comment by lajosrodek** [comment]
And one more: Please add an UT case for the above scenario!

File: core/src/test/java/com/sequenceiq/distrox/v1/distrox/converter/InstanceTemplateParameterConverterTest.java [hunk 2]
```diff
@@ -198,6 +198,15 @@ void convertTestAwsInstanceTemplateV1ParametersToAwsInstanceTemplateV4Parameters
...
     }
 
+    @Test
+    void testWhenDatahubEncryptionTypeSetToNull() {
+        AwsInstanceTemplateV1Parameters source = new AwsInstanceTemplateV1Parameters();
+        source.setEncryption(encryption(null, null));
+        DetailedEnvironmentResponse environment = createDetailedEnvironmentResponseForAwsEncryption(true, false, null);
+        AwsInstanceTemplateV4Parameters awsInstanceTemplateV4Parameters = underTest.convert(source, environment);
+        assertThat(awsInstanceTemplateV4Parameters).isNotNull();
+    }
```
Comment (lajosrodek) [inline core/src/test/java/com/sequenceiq/distrox/v1/distrox/converter/InstanceTemplateParameterConverterTest.java:208]: Please also assert that:
* `awsInstanceTemplateV4Parameters.getEncryption()` is not `null`.
* `awsInstanceTemplateV4Parameters.getEncryption().getKey()` and `awsInstanceTemplateV4Parameters.getEncryption().getType()` are both `null`.

**Review by lajosrodek — CHANGES_REQUESTED** [review]
One minor ask.

File: core/src/test/java/com/sequenceiq/distrox/v1/distrox/converter/InstanceTemplateParameterConverterTest.java (hunk 2 above)
Comment (satyabollineni) [reply core/src/test/java/com/sequenceiq/distrox/v1/distrox/converter/InstanceTemplateParameterConverterTest.java:212]: Addressed this.

**Review by lajosrodek — APPROVED** [review]
Thanks, LGTM

**Comment by lajosrodek** [comment]
Actually, I am not sure if `master` is the right branch. The Jira ticket mentions 2.49 & 2.50 as the target branches, in which case this should be first landing in 2.49 as a HF, then merged forward to 2.50, and from there merged to `master`.

**Comment by lajosrodek** [comment]
Please see my comment https://github.com/hortonworks/cloudbreak/pull/11842#issuecomment-983944308 and repurpose this PR for CB-2.49.0.

**Review by lajosrodek — APPROVED** [review]
Still okay

**Pull request merged**